pkg/build/image/image_tree.go

```diff
@@ -203,6 +208,8 @@ func (tree *ImagesTree) GetMultiplatformImage(name string) *MultiplatformImage {
 }
 
 func (tree *ImagesTree) SetMultiplatformImage(newImg *MultiplatformImage) {
```

Comment:
this appears to be called concurrently here: https://github.com/werf/werf/blob/main/pkg/build/build_phase.go#L268

which I believe is causing a race condition that causes images to be missing in the `multiplatformImages` - adding a mutex should synchronize access to the array and prevent the race